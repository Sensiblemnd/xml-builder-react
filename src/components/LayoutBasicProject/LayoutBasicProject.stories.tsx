import * as React from "react";

import { storiesOf } from "@storybook/react";
import LayoutBasicProject from "./LayoutBasicProject";

import {
  PageSection,
  Bullseye,
  EmptyState,
  EmptyStateVariant,
  Title,
  EmptyStateIcon,
  Nav,
  NavItem,
  NavGroup
} from "@patternfly/react-core";
import { MusicIcon } from "@patternfly/react-icons";

storiesOf("Project. Layout Basic", module).add("Default", () => (
  <LayoutBasicProject
    version="1.0.0"
    productName="Mi producto"
    githubSourceCodeURL="https://github.com/project-openubl/xml-builder"
    githubIssuesURL="https://github.com/project-openubl/xml-builder/issues"
    documentationURL="https://project-openubl.gitbook.io/xml-builder/"
    swaggerApiURL="https://app.swaggerhub.com/organizations/project-openubl"
    sidebar={
      <Nav id="nav-primary-simple" aria-label="Nav" theme="dark">
        <NavGroup title="Group">
          <NavItem className="pf-m-current">My menu</NavItem>
        </NavGroup>
      </Nav>
    }
  >
    <PageSection>
      <Bullseye>
        <EmptyState variant={EmptyStateVariant.full}>
          <EmptyStateIcon icon={MusicIcon} />
          <Title headingLevel="h5" size="lg">
            Your content
          </Title>
        </EmptyState>
      </Bullseye>
    </PageSection>
  </LayoutBasicProject>
));
