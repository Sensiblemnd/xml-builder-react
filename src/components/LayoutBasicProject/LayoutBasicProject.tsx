import * as React from "react";
import { Page, PageSidebar, SkipToContent } from "@patternfly/react-core";
import { ButtonAboutProject } from "../ButtonAboutProject";
import { HeaderProject } from "../HeaderProject";

export interface LayoutBasicProjectProps {
  version: string;
  productName: string;
  swaggerApiURL: string;
  githubIssuesURL: string;
  documentationURL: string;
  githubSourceCodeURL: string;
  sidebar: React.ReactNode;
}

interface State {}

export class LayoutBasicProject extends React.Component<
  LayoutBasicProjectProps,
  State
> {
  renderPageSkipToContent = () => {
    return (
      <SkipToContent href="#primary-app-container">
        Skip to Content
      </SkipToContent>
    );
  };

  renderAboutButton = () => {
    const {
      version,
      productName,
      swaggerApiURL,
      githubIssuesURL,
      documentationURL,
      githubSourceCodeURL
    } = this.props;

    return (
      <ButtonAboutProject
        version={version}
        productName={productName}
        githubSourceCodeURL={githubSourceCodeURL}
        githubIssuesURL={githubIssuesURL}
        documentationURL={documentationURL}
        swaggerApiURL={swaggerApiURL}
      />
    );
  };

  render() {
    const { sidebar, children } = this.props;

    return (
      <React.Fragment>
        <Page
          header={<HeaderProject aboutButton={this.renderAboutButton()} />}
          sidebar={<PageSidebar nav={sidebar} theme="dark" />}
          isManagedSidebar
          skipToContent={this.renderPageSkipToContent()}
        >
          {children}
        </Page>
      </React.Fragment>
    );
  }
}
