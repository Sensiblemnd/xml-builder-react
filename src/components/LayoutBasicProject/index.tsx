import LayoutBasicProject from "./LayoutBasicProject";

export default LayoutBasicProject;
