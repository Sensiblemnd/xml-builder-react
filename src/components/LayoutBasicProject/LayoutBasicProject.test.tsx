import React from "react";
import { shallow } from "enzyme";
import LayoutBasicProject from "./LayoutBasicProject";

it("Renders without crashing", () => {
  const wrapper = shallow(
    <LayoutBasicProject
      version="1.0.0"
      productName="Mi producto"
      githubSourceCodeURL="https://github.com/project-openubl/xml-builder"
      githubIssuesURL="https://github.com/project-openubl/xml-builder/issues"
      documentationURL="https://project-openubl.gitbook.io/xml-builder/"
      swaggerApiURL="https://app.swaggerhub.com/organizations/project-openubl"
    />
  );

  expect(wrapper).toMatchSnapshot();
});
