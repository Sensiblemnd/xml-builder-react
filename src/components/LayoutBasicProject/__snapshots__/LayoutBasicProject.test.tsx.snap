// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing 1`] = `
<Fragment>
  <Page
    breadcrumb={null}
    className=""
    defaultManagedSidebarIsOpen={true}
    header={
      <HeaderProject
        aboutButton={
          <ButtonAboutProject
            documentationURL="https://project-openubl.gitbook.io/xml-builder/"
            githubIssuesURL="https://github.com/project-openubl/xml-builder/issues"
            githubSourceCodeURL="https://github.com/project-openubl/xml-builder"
            productName="Mi producto"
            swaggerApiURL="https://app.swaggerhub.com/organizations/project-openubl"
            version="1.0.0"
          />
        }
      />
    }
    isManagedSidebar={true}
    mainContainerId={null}
    onPageResize={[Function]}
    sidebar={
      <PageSidebar
        theme="dark"
      />
    }
    skipToContent={
      <SkipToContent
        className=""
        component="a"
        href="#primary-app-container"
        show={false}
      >
        Skip to Content
      </SkipToContent>
    }
  />
</Fragment>
`;
